wagtail/core/models.py

```diff
@@ -3310,9 +3312,11 @@ def __init__(self, user_perms, page):
     def user_has_lock(self):
         return self.page.locked_by_id == self.user.pk
 
+    @cached_property
```

Comment:
question: does this fall under the same problem area as https://github.com/wagtail/wagtail/pull/6878#pullrequestreview-604997462 ? cc @jacobtoppm

Reply:
I believe so, yes. At least I'm thinking about the scenario where someone approves a page (not locked) and it moves into another task where the page is now locked for them, and I'm not sure this would behave correctly in that case without clearing the cached property.